components/suggest/src/db.rs

```diff
@@ -429,12 +432,24 @@ impl<'a> SuggestDao<'a> {
                 LIMIT
                   :suggestions_limit
                 "#,
-                named_params! {
-                    ":keyword_prefix": keyword_prefix,
-                    ":keyword_suffix": keyword_suffix,
-                    ":provider": SuggestionProvider::Amo,
-                    ":suggestions_limit": suggestions_limit,
-                },
+            &[
+                (":keyword_prefix", &keyword_prefix as &dyn ToSql),
+                (":keyword_suffix", &keyword_suffix as &dyn ToSql),
+                (":provider", provider as &dyn ToSql),
+                (":suggestions_limit", &suggestions_limit as &dyn ToSql),
+            ],
+            mapper,
```

Comment:
actually how do you do that? 😅 
query_rows_and_then_cached takes in mapper with `FnMut(&Row<'_>) -> Result<T, E>`

Reply:
You need to make a closure than inputs just the row, then passes all the data to the mapper function.

(Also you then need to make the `mapper` variable mutable since you're now calling it rather than passing it to `query_rows_and_then`).

I think something like this should do the trick: https://gist.github.com/bendk/9b28f66d6451ffcc8922c470cd630326